eng/pipelines/templates/jobs/ci.yml

```diff
@@ -47,301 +47,82 @@ parameters:
     default: false
 
 jobs:
-  - job: Build
-    pool:
-      name: $(WINDOWSPOOL)
-      image: $(WINDOWSVMIMAGE)
-      os: windows
-
-    variables:
-      Codeql.Enabled: true
-      Codeql.BuildIdentifier: ${{ parameters.ServiceDirectory }}
-      Codeql.SkipTaskAutoInjection: false
-      SDKType: ${{ parameters.SDKType }}
-
-    steps:
-      - ${{ if not(contains(variables['Build.DefinitionName'], '-pr - ')) }}:
-        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
-          parameters:
-            Paths:
-              - "/*"
-              - "!SessionRecords"
-      - template: /eng/pipelines/templates/steps/install-dotnet.yml
-      - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
-        parameters:
-          PackageName: "Azure.Template"
-          ServiceDirectory: "template"
+  - ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
+    - template: /eng/common/pipelines/templates/jobs/generate-job-matrix.yml
+      parameters:
+        GenerateJobName: generate_build_matrix
+        JobTemplatePath: /eng/pipelines/templates/jobs/batched-build-analyze.yml
+        MatrixConfigs:
+        - Name: NET_ci_build_base
+          Path: eng/pipelines/templates/stages/build-matrix.json
+          Selection: sparse
+          GenerateVMJobs: true
+        CloudConfig:
```

Comment:
I don't think CloudConfig is interesting here.

Reply:
I think you're right that it's unnecessary.

Reply:
I'm going to try removing it, but I'm a bit leery that it's actually going to shatter because of the default filter accessing it. Maybe not? We'll see.